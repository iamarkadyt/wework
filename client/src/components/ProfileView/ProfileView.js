import React, { Component } from 'react'
import './ProfileView.scss'
import { shape, string, func, arrayOf } from 'prop-types'
import { followerType, profileType } from '../../types/index'
import { connect } from 'react-redux'
import { deleteExperience, deleteEducation } from '../../state/actions/profileActions'
import { followAPerson, unfollowAPerson } from '../../state/actions/userActions'
import Bio from './Bio/Bio'
import DangerZone from './DangerZone/DangerZone'
import Education from './Education/Education'
import Experience from './Experience/Experience'
import Overview from './Overview/Overview'
import Skills from './Skills/Skills'

class ProfileView extends Component {
  defaultDeletingState = {
    deletingExpEntries: false,
    deletingEduEntries: false
  }

  constructor(props) {
    super(props)
    this.state = {
        ...this.defaultDeletingState
    }
  }

  quitEntryDeletingMode() {
    this.setState(this.defaultDeletingState)
  }

  setExpDeleting(bool) {
    this.setState({
      deletingExpEntries: bool
    })
  }

  setEduDeleting(bool) {
    this.setState({
      deletingEduEntries: bool
    })
  }

  navTo(where) {
    const {
      history,
      match
    } = this.props

    const baseUrl = match.url || ''
    return history.push(`${baseUrl}/${where}`)
  }

  isFollowingProfileOwner() {
    const { 
      profile: {
        user: {
          _id
        }
      },
      authedUser: {
        following
      }
    } = this.props

    return !!following.find(item => item.user === _id)
  }

  // what a bad naming! wheres your <is>???
  profileBelongsToAuthedUser() {
    const { 
      profile: {
        user: {
          _id
        }
      },
      authedUser: {
        id: authedUserId
      }
    } = this.props

    return authedUserId === _id
  }

  render() {
    const {
      profile: {
        bio,
        skills,
        experience,
        education
      },
      profile,
      deleteExperience,
      deleteEducation,
      followAPerson,
      unfollowAPerson
    } = this.props

    const commonProps = {
      navTo: this.navTo.bind(this),
      profileBelongsToAuthedUser: this.profileBelongsToAuthedUser(),
      quitEntryDeletingMode: this.quitEntryDeletingMode.bind(this)
    }

    return (
      <div className="ProfileView-container">
        <Overview 
          profile={profile} 
          isFollowingProfileOwner={this.isFollowingProfileOwner()}
          unfollowAPerson={unfollowAPerson}
          followAPerson={followAPerson} 
          {...commonProps} />
        <Bio bio={bio} />
        <Skills skills={skills} />
        <Experience 
          experience={experience}
          deleteExperience={deleteExperience}
          isDeleting={this.state.deletingExpEntries}
          setDeleting={this.setExpDeleting.bind(this)}
          {...commonProps} />
        <Education 
          education={education} 
          deleteEducation={deleteEducation}
          isDeleting={this.state.deletingEduEntries}
          setDeleting={this.setEduDeleting.bind(this)}
          {...commonProps} />
        <DangerZone 
          {...commonProps} />
      </div>
    )
  }
}

ProfileView.propTypes = {
  authedUser: shape({
    id: string.isRequired,
    following: arrayOf(followerType).isRequired
  }).isRequired,
  history: shape({
    push: func.isRequired
  }).isRequired,
  match: shape({
    url: string.isRequired
  }).isRequired,
  deleteEducation: func.isRequired,
  deleteExperience: func.isRequired,
  followAPerson: func.isRequired,
  unfollowAPerson: func.isRequired,
  profile: profileType.isRequired             
}

export const mapStateToProps = state => ({
  authedUser: state.user
})

export { ProfileView as _UnconnectedProfileView }
export default connect(mapStateToProps, { 
  deleteEducation, 
  deleteExperience, 
  followAPerson, 
  unfollowAPerson 
})(ProfileView)
