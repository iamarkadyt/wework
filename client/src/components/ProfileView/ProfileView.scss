.ProfileView-container {
    display: flex;
    flex-direction: column;
    background-color: white;
    border-radius: .3rem;
    overflow: hidden;
    box-shadow: 0px 3px 23px 0px rgba(0, 0, 0, 0.5);
    section {
        margin: 0 0 1.5rem 0;
        padding: .5rem 2rem;
        display: grid;
        grid-template-columns: .2fr 1fr;
        grid-column-gap: .2rem;
        justify-items: start;
        .header {
            font-weight: bolder;
            font-size: 1.2rem;
        }
    }
    .intro {
        padding: 1.5rem 2rem;
        grid-column-gap: 1rem;
        grid-template-columns: .46fr 1fr .1fr;
        .info {
            .links {
                margin-top: .4rem;
                a {
                    color: white;
                    font-size: 1.5rem;
                    margin-right: .4rem;
                    &:hover {
                        color: rgba(0, 0, 0, 0.5);
                    }
                }
            }
            color: white;
        }
        img {
            width: 10rem;
        }
        background-color: cornflowerblue;
    }
    .skills {
        .tag {
            background-color: lightgray;
            border-radius: .1rem;
            margin-right: .2rem;
            padding: .2rem 1rem;
        }
    }
    .experience,
    .education {
        .nodes {
            .node {
                &:not(:last-child) {
                    margin-bottom: 1rem;
                }
            }
        }
    }
}