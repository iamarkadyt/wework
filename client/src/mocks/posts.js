export const mockPost = {
    "_id": "c717a50ac29bacb2ee",
    "user": {
        "_id": "some_alphanumeric_gibberish",
        "name": "Dwayne Johnson",
        "avatar": ""
    },
    "text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
    "comments": [
        {
            "user": {
                "_id": "some_alphanumeric_gibberish2",
                "name": "Vin Diesel",
                "avatar": ""
            },
            "text": "Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
            "date": "1535133201"
        },
        {
            "user": {
                "_id": "some_alphanumeric_gibberish2",
                "name": "Vin Diesel",
                "avatar": ""
            },
            "text": "Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
            "date": "1535133201"
        },
        {
            "user": {
                "_id": "some_alphanumeric_gibberish2",
                "name": "Vin Diesel",
                "avatar": ""
            },
            "text": "Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
            "date": "1535133201"
        }
    ],
    "likes": [
        {
            "_id": "19825918265102965",
            "user": "aoeu9a8o10s392h0"
        },
        {
            "_id": "19825918265102966",
            "user": "aoeu9a8o10s392hx0"
        },
        {
            "_id": "19825918265102967",
            "user": "aoeu9a8o10s392120"
        }
    ],
    "date": "1535133024"
}

export default new Array(10).fill(mockPost, 0, 10)
