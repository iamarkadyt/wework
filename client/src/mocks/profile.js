export default {
    "skills": [
        "Assembler",
        "Go",
        "C#",
        "C++"
    ],
    "social": [],
    "date": "2018-08-23T20:25:12.263Z",
    "_id": "5b7b42ba57a5806b896e8f0d",
    "user": {
        "_id": "5b75f023f1957052e52106f5",
        "name": "John Doe"
    },
    "__v": 0,
    "handle": "techguy_handle",
    "status": "Actively applying",
    "title": "Senior Developer",
    "company": "Itron",
    "experience": [
        {
            "current": false,
            "_id": "5b7ca6cf63353349cf03756c",
            "title": "SWE",
            "company": "Revature",
            "from": "2009-03-31T23:00:00.000Z",
            "to": "2025-08-01T01:13:50.111Z"
        }
    ],
    "education": [
        {
            "current": true,
            "_id": "5b7cab633345cc4c066c0540",
            "school": "AC Bootcamp",
            "degree": "Nano CS Degree",
            "from": "2015-08-02T18:33:20.000Z"
        }
    ],
    "bio": "My name is [techguyinfo]. I am a senior developer currently working at Microsoft. I specialize in AI and Cloud computing.",
    "githubusername": "arkadyt",
    "location": "Seattle, WA",
    "website": "techguy.info"
}