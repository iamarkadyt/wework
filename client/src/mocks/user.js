export default {
  id: "98ac61abed72ee8d12"
}
