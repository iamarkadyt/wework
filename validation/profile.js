const Yup = require('yup')
const Profile = require('../models/Profile')

/**
 * Check to see if value (by its pathname) exists in the database.
 * 'user' ObjectID is required to see if it belongs to the current user,
 * in which case there is no error --> user is updating his profile. 
 */
Yup.addMethod(Yup.mixed, 'unoccupied', function (user, message) {
    return this.test('unoccupied', message, function (value) {
        const { path, createError } = this;
        return new Promise((resolve, reject) => {
            // check to see if any profile has already taken the value
            Profile.findOne({ [path]: value }).then(profile => {
                // compare mongoose's ObjectIDs with mongoose's .equals()
                if (profile && !profile.user.equals(user))
                    reject(createError({ path, message }))
                // else value is free, or it belongs to the current user
                else resolve(true)
            })
        })
    })
})

module.exports = (profile, options = {
    onlyEducation: false,
    onlyExperience: false
}) => new Promise((resolve, reject) => {
    let schema = {}

    if (options.onlyExperience) {
        schema = Yup.object().shape({
            title: Yup.string().required("Please specify your title"),
            company: Yup.string().required("Please enter the company name"),
            location: Yup.string(),
            from: Yup.string().required("Please specify the start date"),
            to: Yup.string().when('current', {
                is: undefined,
                then: Yup.string().required("Do you still work here?"),
                otherwise: Yup.string().notRequired()
            }),
            current: Yup.bool(),
            description: Yup.string().min(300, "Please enter at least 300 characters")
        })
    } else if (options.onlyEducation) {
        schema = Yup.object().shape({
            school: Yup.string().required("Please enter a school name"),
            degree: Yup.string().required("Please specify the degree type"),
            fieldOfStudy: Yup.string().required("Please specify your field of study"),
            from: Yup.string().required("Please enter the start date"),
            to: Yup.string().when('current', {
                is: undefined,
                then: Yup.string().required("Do you still study here?"),
                otherwise: Yup.string().notRequired()
            }),
            current: Yup.bool(),
            description: Yup.string().min(300, "Minimum of 300 characters is required!")
        })
    } else { /* For whole profile with the exclusion of above described fields */
        schema = Yup.object().shape({
            handle: Yup.string()
                .required('Please enter your profile handle')
                .min(2, 'A minimum of 2 characters is required')
                .max(40, 'Maximum allowed characters is 40')
                .unoccupied(profile.user, 'Handle is already taken'),
            status: Yup.string()
                .required('Please specify your job seeker status'),
            title: Yup.string()
                .required('Please specify your job title'),
            skills: Yup.array().of(Yup.string().typeError('Elements must be strings'))
                .min(3, 'Please enter at least ${min} skills')
                .max(40, 'Skills limit of ${max} has been reached')
                .required('Please specify your skills'),
            youtube: Yup.string().url('Please enter a valid URL address'),
            twitter: Yup.string().url('Please enter a valid URL address'),
            facebook: Yup.string().url('Please enter a valid URL address'),
            linkedin: Yup.string().url('Please enter a valid URL address'),
            instagram: Yup.string().url('Please enter a valid URL address'),
            githubusername: Yup.string()
                .matches(/^[a-z\d](?:[a-z\d]|-(?=[a-z\d])){0,38}$/i,
                    'Please enter a valid github username')
        })
    }

    // strict flag ensures that value coercion and object transformation would not happen
    // this way db stays clean of empty fields 
    schema.validate(profile, { abortEarly: false, strict: true })
        .then(profile => resolve(profile))
        .catch(err => {
            const errors = {}
            for (let i of err.inner)
                errors[i.path] = i.message
            reject(errors)
        })
})