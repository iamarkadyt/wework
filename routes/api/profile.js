const express = require('express')
const router = express.Router()
const mongoose = require('mongoose')
const passport = require('passport')
const Profile = require('../../models/Profile')
const User = require('../../models/User')

// @route   GET api/profile/test
// @desc    Tests profile route.
// @access  Public
router.get('/test', (req, res) => {
    res.json({ message: 'OK' })
})


// @route   GET api/profile
// @desc    Get current users profile
// @access  Private
router.get('/', passport.authenticate('jwt', { session: false }), (req, res) => {
    const errors = {}

    Profile.findOne({ user: req.user.id })
        .then(profile => {
            if (!profile) {
                errors.noprofile = 'There is no profile for this user'
                res.status(404).json({ errors })
            }

            res.json(profile)
        })
        .catch(err => res.status(404).json(err))
})

// @route   POST /api/profile
// @desc    Create or Update user profile
// @access  Protected
router.post('/', passport.authenticate('jwt', { session: false }), (req, res) => {
    require('../../validation/profile')({ ...req.body, user: req.user.id })
        .then(profile => {
            Profile.findOneAndUpdate(
                { user: req.body.id },
                { $set: profile }, // form fields left empty must not be sent
                { new: true, upsert: true })
                .then(profile => res.json(profile))
                .catch(err => res.status(400).json(err))
        })
        .catch(errors => res.status(400).json(errors))
})


module.exports = router;