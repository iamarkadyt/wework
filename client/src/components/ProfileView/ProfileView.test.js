import React from 'react'
import { shallow } from 'enzyme'
import { _UnconnectedProfileView as ProfileView, mapStateToProps } from './ProfileView'
import cloneDeep from 'lodash.clonedeep'
import { mockProfile as profile } from '../../mocks/profile'

let mountedComponent, props

const getMockProps = () => {
  return cloneDeep({
    authedUser: {
      id: 'someId71926',
      following: [ 
        { _id: 'someId1101282', user: 'userId34491f0' }
      ]
    },
    history: { push: jest.fn() },
    match: { url: '' },
    deleteEducation: jest.fn(),
    deleteExperience: jest.fn(),
    followAPerson: jest.fn(),
    unfollowAPerson: jest.fn(),
    profile,
  })
}

const comp = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<ProfileView {...props} />)
  }
  return mountedComponent
}

describe('ProfileView', () => {
  beforeEach(() => {
    props = getMockProps()
    mountedComponent = undefined
  })

  describe('functions', () => {
    describe('render', () => {
      it('calls isFollowingProfileOwner', () => {
        const spy = jest.spyOn(ProfileView.prototype, 'isFollowingProfileOwner')
        comp()
        expect(spy).toHaveBeenCalled()
      })

      it('calls profileBelongsToAuthedUser', () => {
        const spy = jest.spyOn(ProfileView.prototype, 'profileBelongsToAuthedUser')
        comp()
        expect(spy).toHaveBeenCalled()
      })
    })

    describe('navTo', () => {
      it('calls history.push', () => {
        const mockFn = jest.fn()
        props.history.push = mockFn
        comp().instance().navTo()
        expect(mockFn).toHaveBeenCalled()
      })

      it('calls history.push with correct args', () => {
        const mockFn = jest.fn()
        props.history.push = mockFn

        const link = 'link'
        props.match.url = 'URL'

        comp().instance().navTo(link)
        expect(mockFn).toHaveBeenCalledWith(`${props.match.url}/${link}`)
      })
    })

    describe('setEduDeleting', () => {
      it('sets respective var in the state correctly', () => {
        comp().setState({ deletingEduEntries: true })
        comp().instance().setEduDeleting(false)
        expect(comp().state('deletingEduEntries')).toBe(false)
      })
    })

    describe('setExpDeleting', () => {
      it('sets respective var in the state correctly', () => {
        comp().setState({ deletingExpEntries: true })
        comp().instance().setExpDeleting(false)
        expect(comp().state('deletingExpEntries')).toBe(false)
      })
    })

    describe('quitEntryDeletingMode', () => {
      it('pushes instance().defaultDeletingState to the state', () => {
        const name = 'someCrazyVarName'
        comp().instance().defaultDeletingState = {
          [name]: true
        }
        comp().instance().quitEntryDeletingMode()
        expect(comp().state(name)).toBeDefined()
      })
    })

    describe('defaultDeletingState', () => {
      // yup. some would assume mirroring it like this an antipattern.
      it('is correct', () => {
        const expected = {
          deletingExpEntries: false,
          deletingEduEntries: false
        }
        expect(comp().instance().defaultDeletingState).toStrictEqual(expected)
      })
    })

    describe('constructor', () => {
      it('sets default state', () => {
        expect(comp().state()).toStrictEqual(comp().instance().defaultDeletingState)
      })
    })

    describe('isFollowingProfileOwner', () => {
      it('returns true if owner is followed', () => {
        const sameId = 'id00013'
        props.profile.user._id = sameId
        props.authedUser.following[0].user = sameId
        expect(comp().instance().isFollowingProfileOwner()).toBe(true)
      })

      it('returns false if owner is not followed', () => {
        const someId = 'id00013'
        props.profile.user._id = someId
        props.authedUser.following[0].user = someId + 'a change'
        expect(comp().instance().isFollowingProfileOwner()).toBe(false)
      })
    })

    describe('profileBelongsToAuthedUser', () => {
      it('returns true if profile is owned by authed user', () => {
        const sameId = 'id32221'
        props.profile.user._id = sameId
        props.authedUser.id = sameId
        expect(comp().instance().profileBelongsToAuthedUser()).toBe(true)
      })

      it('returns false if profile is not owned by authed user', () => {
        const someId = 'id32221'
        props.profile.user._id = someId
        props.authedUser.id = someId + 'a change'
        expect(comp().instance().profileBelongsToAuthedUser()).toBe(false)
      })
    })
  })

  describe('props', () => {
    describe('passing', () => {
      describe('Overview', () => {
        it('receives correct thing into unfollowAPerson prop', () => {
          expect(comp().find('overview').prop('unfollowAPerson')).toBe(props.unfollowAPerson)
        })

        it('receives correct thing into followAPerson prop', () => {
          expect(comp().find('overview').prop('followAPerson')).toBe(props.followAPerson)
        })

        it('receives correct thing into navTo prop', () => {
          expect(comp().find('overview').prop('navTo').name).toContain('navTo')
        })

        it('receives correct thing into quitEntryDeletingMode prop', () => {
          expect(comp().find('overview').prop('quitEntryDeletingMode').name).toContain('quitEntryDeletingMode')
        })
      })

      describe('Experience', () => {
        it('receives correct thing into deleteExperience prop', () => {
          expect(comp().find('experience').prop('deleteExperience')).toBe(props.deleteExperience)
        })

        it('receives correct thing into setDeleting prop', () => {
          expect(comp().find('experience').prop('setDeleting').name).toContain('setExpDeleting')
        })

        it('receives correct thing into navTo prop', () => {
          expect(comp().find('experience').prop('navTo').name).toContain('navTo')
        })

        it('receives correct thing into quitEntryDeletingMode prop', () => {
          expect(comp().find('experience').prop('quitEntryDeletingMode').name).toContain('quitEntryDeletingMode')
        })
      })

      describe('Education', () => {
        it('receives correct thing into deleteEducation prop', () => {
          expect(comp().find('education').prop('deleteEducation')).toBe(props.deleteEducation)
        })

        it('receives correct thing into setDeleting prop', () => {
          expect(comp().find('education').prop('setDeleting').name).toContain('setEduDeleting')
        })

        it('receives correct thing into navTo prop', () => {
          expect(comp().find('education').prop('navTo').name).toContain('navTo')
        })

        it('receives correct thing into quitEntryDeletingMode prop', () => {
          expect(comp().find('education').prop('quitEntryDeletingMode').name).toContain('quitEntryDeletingMode')
        })
      })

      describe('DangerZone', () => {
        it('receives correct thing into navTo prop', () => {
          expect(comp().find('dangerZone').prop('navTo').name).toContain('navTo')
        })

        it('receives correct thing into quitEntryDeletingMode prop', () => {
          expect(comp().find('dangerZone').prop('quitEntryDeletingMode').name).toContain('quitEntryDeletingMode')
        })
      })

      describe('common props obj', () => {
        it('contains correct navTo prop', () => {
          // hey you know what? do that per every component.
          // makes sense and you get numbers.
        })
      })
    })
  })

  describe('mapStateToProps', () => {
    it('returns expected object', () => {
      const state = {
        user: {
          name: 'John'
        }
      }
      expect(mapStateToProps(state)).toEqual({ authedUser: state.user })
    })
  })

  it('matches snapshot', () => {
    expect(comp()).toMatchSnapshot()
  })
})
