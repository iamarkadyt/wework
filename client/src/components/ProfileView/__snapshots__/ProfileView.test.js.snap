// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileView matches snapshot 1`] = `
<div
  className="ProfileView-container"
>
  <overview
    followAPerson={[MockFunction]}
    isFollowingProfileOwner={false}
    navTo={[Function]}
    profile={
      Object {
        "__v": 0,
        "_id": "5b7b42ba57a5806b896e8f0d",
        "bio": "TEST BIO TEXT.",
        "company": "Some Company",
        "date": "2018-08-23T20:25:12.263Z",
        "education": Array [
          Object {
            "_id": "5b7cab633345cc4c066c0540",
            "current": true,
            "degree": "Nano CS Degree",
            "fieldOfStudy": "Bioengineering",
            "from": "2015-08-02T00:00:00.000Z",
            "school": "Some Bootcamp",
          },
        ],
        "experience": Array [
          Object {
            "_id": "5b7ca6cf63353349cf03756c",
            "company": "Some Company",
            "current": false,
            "from": "2009-03-31T23:00:00.000Z",
            "title": "Software Developer",
            "to": "2012-08-01T00:00:00.000Z",
          },
        ],
        "githubusername": "arkadyt",
        "handle": "john_doe",
        "location": "Seattle, WA",
        "skills": Array [
          "Go",
          "C#",
          "C++",
          "Whatnot else",
        ],
        "status": "Not available for employment",
        "title": "Senior Developer",
        "user": Object {
          "_id": "5b75f023f1957052e52106f5",
          "avatar": "gravatar://some.avatar",
          "name": "John Doe",
        },
        "website": "arkadyt.com",
      }
    }
    profileBelongsToAuthedUser={false}
    quitEntryDeletingMode={[Function]}
    unfollowAPerson={[MockFunction]}
  />
  <bio
    bio="TEST BIO TEXT."
  />
  <skills
    skills={
      Array [
        "Go",
        "C#",
        "C++",
        "Whatnot else",
      ]
    }
  />
  <experience
    deleteExperience={[MockFunction]}
    experience={
      Array [
        Object {
          "_id": "5b7ca6cf63353349cf03756c",
          "company": "Some Company",
          "current": false,
          "from": "2009-03-31T23:00:00.000Z",
          "title": "Software Developer",
          "to": "2012-08-01T00:00:00.000Z",
        },
      ]
    }
    isDeleting={false}
    navTo={[Function]}
    profileBelongsToAuthedUser={false}
    quitEntryDeletingMode={[Function]}
    setDeleting={[Function]}
  />
  <education
    deleteEducation={[MockFunction]}
    education={
      Array [
        Object {
          "_id": "5b7cab633345cc4c066c0540",
          "current": true,
          "degree": "Nano CS Degree",
          "fieldOfStudy": "Bioengineering",
          "from": "2015-08-02T00:00:00.000Z",
          "school": "Some Bootcamp",
        },
      ]
    }
    isDeleting={false}
    navTo={[Function]}
    profileBelongsToAuthedUser={false}
    quitEntryDeletingMode={[Function]}
    setDeleting={[Function]}
  />
  <dangerZone
    navTo={[Function]}
    profileBelongsToAuthedUser={false}
    quitEntryDeletingMode={[Function]}
  />
</div>
`;
